File: Progetto-2023-2024-Vecchio-Larocca-Lentini-Mahrous/backend/src/main/java/it/unical/inf/ea/backend/dto/basics/UserBasicDTO.java
package it.unical.inf.ea.backend.dto.basics;

import it.unical.inf.ea.backend.dto.UserImageDTO;
import it.unical.inf.ea.backend.dto.enums.Provider;
import it.unical.inf.ea.backend.dto.enums.UserRole;
import it.unical.inf.ea.backend.dto.enums.UserStatus;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotNull;
import lombok.*;
import org.hibernate.validator.constraints.Length;

@Data
@NoArgsConstructor
@ToString
@AllArgsConstructor
@Builder
public class UserBasicDTO {

    @NotNull
    private String id;

    @NotNull
    private String lastName;

    @NotNull
    private String firstName;

    @Email
    @NotNull
    private String email;

    private String phoneNumber;

    private UserImageDTO image;

    private UserStatus status;

    private Provider provider;

    private UserRole role;

}

File: Progetto-2023-2024-Vecchio-Larocca-Lentini-Mahrous/frontend/app/src/main/java/com/android/frontend/dto/basic/UserBasicDTO.kt
package com.android.frontend.dto.basic

import com.android.frontend.dto.UserDTO
import com.android.frontend.dto.UserImageDTO

data class UserBasicDTO(
    var id: String,
    var lastName: String,
    var firstName: String,
    var email: String,
    var phoneNumber: String,
    var photoProfile: UserImageDTO,
    var status: UserDTO.UserStatus,
    var role: UserDTO.UserRole
)

